### DISCUSSION TITLE
Reversing Clipping Plane Direction in FusionClipMethod for pyAnatomist

### DISCUSSION THREAD

**Discussion by augustin-ogier** [brainvisa/brainvisa.github.io#178] [Q&A]
Hi, I would like to know if it is possible to reverse the clipping of a FusionClipMethod between a mesh and a slice. Specifically, is there a way to choose which side of the plane is clipped and not displayed? Using the graphical interface or preferably pyAnatomist

For example, in this visualisation, I'd like the mesh to be below the axial slice and not above it.
![image](https://github.com/brainvisa/brainvisa.github.io/assets/104825537/91c2b0ac-5b75-4027-9c4b-ba033782acb7)

Thanks!

**Comment by denisri** [comment]
Hi,
Oh you're right, this is something missing in Anatomist. Ii exists for the window slice (oblique control) but not for sliced objects (cut control). Now that you say it I have just implemented it in the development version (master branches, to be built fro sources).
It will be available as Ctrl+I key when the cut control is active (same as in the oblique control).
In the pyanatomist API, it will be:
```
# assuming "object" is an anatomist object with a slice inside (inherits SelfSliceable)
object.invertSlice()
object.notifyObservers()
```
If you don't compile the development version, in the meantime you can do the same a bit more manually in the released version:
```
import aims
import numpy as np

q = aims.Quaternion()
q.fromAxis((1, 0, 0), np.pi)  # rotate pi rad around X axis
# rotate the object current quat
object.setQuaternion(object.quaternion() * q)
object.notifyObservers()
```
Denis

**Reply by augustin-ogier** [reply]
Hi, thanks a lot for the detailled answer! 
Following the code you gave, I currently have the error `AttributeError: module 'aims' has no attribute 'Quaternion'.` 
I'm working with the virtualbox brainvisa-5.1.1. Could you tell me if upgrading to 5.1.2 would solve the problem?

**Reply by augustin-ogier** [reply]
I found the solution. I just had to do `from soma import aims` instead of `import aims` directly. 
Thanks again for the prompt help and for all the tools you make available!

Augustin
